samza-core/src/test/java/org/apache/samza/diagnostics/TestDiagnosticsManager.java

```diff
@@ -272,6 +278,7 @@ private void validateOutgoingMessageEnvelope(OutgoingMessageEnvelope outgoingMes
     Assert.assertEquals(containerNumCores, diagnosticsStreamMessage.getContainerNumCores().intValue());
     Assert.assertEquals(numPersistentStores, diagnosticsStreamMessage.getNumPersistentStores().intValue());
     Assert.assertEquals(autosizingEnabled, diagnosticsStreamMessage.getAutosizingEnabled());
+    Assert.assertEquals(config, diagnosticsStreamMessage.getConfig());
```

Comment:
Config does not define an equals method, so not sure how this test is working.

Reply:
This invokes the equals method of MapConfig class.